### PR TITLE
Add `push` and `pull` functions commands

### PR DESCRIPTION
**Pull request by parkerhendo**
## Add `bt functions push` and `bt functions pull`

Adds push/pull commands for managing Braintrust functions as code. Push bundles local TypeScript/Python files and uploads function definitions (code functions, prompts, tools) to Braintrust. Pull materializes prompt definitions from Braintrust back to local files.

#### `bt functions push [PATH...] --if-exists <error|replace|ignore>`

| Flag                        | Short | Default | Description                                                     |
| --------------------------- | ----- | ------- | --------------------------------------------------------------- |
| `--file <PATH>`             | —     | —       | File or directory paths (also accepts positional args)          |
| `--if-exists`               | —     | `error` | Behavior when slug already exists: `error`, `replace`, `ignore` |
| `--language`                | —     | `auto`  | Force runtime: `auto`, `javascript`, `python`                   |
| `--runner <BIN>`            | —     | —       | Override runner binary (e.g. `tsx`, `vite-node`, `python`)      |
| `--tsconfig <PATH>`         | —     | —       | tsconfig path for JS runner and bundler                         |
| `--external-packages <PKG>` | —     | —       | Additional packages to mark external during JS bundling         |
| `--requirements <PATH>`     | —     | —       | Python requirements file                                        |
| `--terminate-on-failure`    | —     | `false` | Stop after first hard failure                                   |
| `--yes`                     | `-y`  | `false` | Skip confirmation prompt                                        |

#### `bt functions pull [SLUG...] --language <typescript|python>`

| Flag                  | Short | Default        | Description                                             |
| --------------------- | ----- | -------------- | ------------------------------------------------------- |
| `--slug`              | `-s`  | —              | Function slug(s) to pull (also accepts positional args) |
| `--output-dir <PATH>` | —     | `./braintrust` | Destination directory for generated files               |
| `--language`          | —     | `typescript`   | Output language: `typescript`, `python`                 |
| `--project-id`        | —     | —              | Scope to a specific project by ID                       |
| `--id`                | —     | —              | Select a specific function by ID                        |
| `--version`           | —     | —              | Version selector                                        |
| `--force`             | —     | `false`        | Overwrite existing files                                |
| `--verbose`           | —     | `false`        | Show skipped/unsupported records                        |

### How it works

**Push** runs a JS/Python runner to discover function and prompt registrations from source files, bundles code functions with esbuild, and uploads definitions to Braintrust via `/insert-functions`. Handles both modern (`toFunctionDefinition`) and legacy (`project.prompts.create()`) prompt registration paths.

**Pull** fetches prompt definitions from a project and writes them as TypeScript or Python source files. Non-prompt function types are skipped.

### Infrastructure

- JS bundler script (esbuild) and runner scripts for TS/Python
- Shared `runner-common.ts` and `python_runner_common.py` modules
- Atomic file writes, git helpers, source language detection, Python interpreter resolution

### Test plan

- [ ] `cargo test --test functions` — 20 integration tests covering push/pull CLI fixtures, mock server round-trips, edge cases
- [ ] Manual: `bt functions push prompt.ts --if-exists replace` pushes both code functions and prompts
- [ ] Manual: `bt functions pull --slug <slug>` materializes prompt to local file

### CODE REVIEW

**Comment by parkerhendo** [comment]
Updated integrations tests here. Had to make a few tweaks to get them to pass, but all are green now.

https://github.com/braintrustdata/braintrust/pull/11810

**Comment by nselvidge** [comment]
@parkerhendo I pulled this branch and had some failing tests. looking at our CI it looks like we don't run all new tests by default, I don't think the new tests are getting run in CI.

File: src/functions/pull.rs [hunk 1]
```diff
@@ -0,0 +1,1913 @@
...
+
+    let mut seen_names = BTreeSet::new();
+    for row in prompts {
+        let var_name = sanitize_renderer_identifier(
+            &row.variable_seed,
+            FunctionsLanguage::Python,
+            &mut seen_names,
+        );
+        out.push_str(&format!("{var_name} = project.prompts.create(\n"));
+        out.push_str(&format!(
```
[the remark this answers is not in the document]
Comment (parkerhendo) [reply src/functions/pull.rs]: Fixed this. Thanks for flagging!
